[FSDP] Add state_dict() save/reload in parity test

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #73532
* #73453
* __->__ #73366

Adds state_dict() save/reload in parity with DDP test to ensure
checkpointing doesn't cause issue with accuracy/model params.

Differential Revision: [D34434358](https://our.internmc.facebook.com/intern/diff/D34434358/)

rohan-varma committed:
[FSDP] Add state_dict() save/reload in parity test

Adds state_dict() save/reload in parity with DDP test to ensure
checkpointing doesn't cause issue with accuracy/model params.

Differential Revision: [D34434358](https://our.internmc.facebook.com/intern/diff/D34434358/)

[ghstack-poisoned]

fegin reviewed (approved):
Thanks for the test.

rohan-varma committed:
Update on "[FSDP] Add state_dict() save/reload in parity test"


Adds state_dict() save/reload in parity with DDP test to ensure
checkpointing doesn't cause issue with accuracy/model params.

Differential Revision: [D34434358](https://our.internmc.facebook.com/intern/diff/D34434358/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[FSDP] Add state_dict() save/reload in parity test"


Adds state_dict() save/reload in parity with DDP test to ensure
checkpointing doesn't cause issue with accuracy/model params.

Differential Revision: [D34434358](https://our.internmc.facebook.com/intern/diff/D34434358/)

[ghstack-poisoned]
